src/k8s-engine/wsl.ts

```diff
@@ -142,7 +142,9 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
 
       this.distroProgress.max = parseInt(response.headers.get('Content-Length') || '0');
       await util.promisify(stream.pipeline)(response.body, progress, writeStream);
-      await util.promisify(fs.rename)(outPath, this.distroFile);
+      // fs.rename uses hard links under the hood, breaks on windows cross-drive renames
+      await util.promisify(fs.copyFile)(outPath, this.distroFile);
+      await util.promisify(fs.unlink)(outPath);
```

Comment:
```suggestion
      await fs.promises.copyFile(outPath, this.distroFile);
      await fs.promises.unlink(outPath);
```

Reply:
I redid the code to attempt `rename` the first time and resort to the above when it fails. Safer, faster, not to mention less chance (albeit unlikely) of running out of disk space on a very strapped VM.